### PR TITLE
Feature/270

### PR DESCRIPTION
**Pull request by fenght**
pagination jobList

### CODE REVIEW

File: platform-api/src/main/java/com/flow/platform/api/controller/JobController.java [hunk 1]
```diff
@@ -148,6 +151,29 @@ public void create(@RequestParam(required = false, defaultValue = "true") boolea
...
+            paths = Lists.newArrayList(path);
+        }
+
+        if (Objects.isNull(pageable) || StringUtil
+            .isNullOrEmptyForItems(String.valueOf(pageable.getPageNumber()),
+                String.valueOf(pageable.getPageSize())) ||
+            (pageable.getPageSize() == 0 || pageable.getPageNumber() == 0)) {
+
+            pageable = new Pageable();
+            pageable.setNumber(1);
```
Comment (fenght) [inline platform-api/src/main/java/com/flow/platform/api/controller/JobController.java]: to constans

File: platform-api/src/main/java/com/flow/platform/api/dao/job/JobDaoImpl.java [hunk 2]
```diff
@@ -76,6 +79,33 @@ protected String getKeyName() {
...
+            List<Object[]> objects = nativeQuery.list();
+            List<Job> jobs = JobConvertUtil.convert(objects);
+            return new Page<>(jobs, pageable.getPageSize(), pageable.getPageNumber(), new TotalSupplier() {
+                @Override
+                public long get() {
+                    NativeQuery nativeQuery = session.createNativeQuery(
+                        JOB_COUNT_QUERY + " WHERE job.session_id IN (:sessionIds) AND nr.node_status=:status")
+                        .setParameter("sessionIds", sessionIds)
+                        .setParameter("status", nodeStatus.getName());
+                    return Long.valueOf(nativeQuery.uniqueResult().toString());
```
Comment (fenght) [inline platform-api/src/main/java/com/flow/platform/api/dao/job/JobDaoImpl.java]: Long.class

File: platform-api/src/main/java/com/flow/platform/api/dao/user/PermissionDaoImpl.java [hunk 3]
```diff
@@ -45,6 +49,25 @@ protected String getKeyName() {
             .list());
     }
 
+    @Override
+    public Page<Integer> list(String action, Pageable pageable) {
```
Comment (fenght) [inline platform-api/src/main/java/com/flow/platform/api/dao/user/PermissionDaoImpl.java]: remove

File: platform-api/src/main/java/com/flow/platform/api/dao/user/UserFlowDao.java [hunk 4]
```diff
@@ -30,11 +32,16 @@
      */
     List<String> listByEmail(String email);
 
+
+    Page<String> listByEmail(String email, Pageable pageable);
```
Comment (fenght) [inline platform-api/src/main/java/com/flow/platform/api/dao/user/UserFlowDao.java]: remove

File: platform-api/src/main/java/com/flow/platform/api/service/job/JobSearchServiceImpl.java [hunk 5]
```diff
@@ -51,6 +57,32 @@
...
+    @Override
+    public Page<Job> search(SearchCondition searchCondition, List<String> paths, Pageable pageable) {
+
+        if (Objects.isNull(searchCondition) || StringUtil
+            .isNullOrEmptyForItems(searchCondition.getCategory(), searchCondition.getCreator(),
+                searchCondition.getBranch(), searchCondition.getKeyword())) {
+
+            return jobService.list(paths, false, pageable);
+
+        } else {
```
Comment (fenght) [inline platform-api/src/main/java/com/flow/platform/api/service/job/JobSearchServiceImpl.java]: remove else

File: platform-api/src/main/java/com/flow/platform/api/service/job/JobSearchServiceImpl.java [hunk 6]
```diff
@@ -51,6 +57,32 @@
         return match(searchCondition, jobs);
     }
 
+    @Override
+    public Page<Job> search(SearchCondition searchCondition, List<String> paths, Pageable pageable) {
+
+        if (Objects.isNull(searchCondition) || StringUtil
```
Comment (fenght) [inline platform-api/src/main/java/com/flow/platform/api/service/job/JobSearchServiceImpl.java]: put condition to class

File: platform-plugin/pom.xml [hunk 7]
```diff
@@ -23,6 +23,12 @@
       <version>${project.version}</version>
       </dependency>
 
+    <dependency>
```
Comment (fenght) [inline platform-plugin/pom.xml]: check

File: platform-api/src/main/java/com/flow/platform/api/dao/job/JobDaoImpl.java [hunk 8]
```diff
@@ -92,48 +128,109 @@ public Job get(BigInteger key) {
...
+    public Page<Job> listByPath(List<String> paths, Pageable pageable) {
+        return execute((Session session) -> {
+
+            String countSelect = "count(*)";
+            String where = "";
+            if (!CollectionUtil.isNullOrEmpty(paths)){
+                where = "job.node_path in (:paths) order by job.created_at desc ";
+            }
+
+            Builder builder = ObjectUtil.deepCopy(QUERY_BUILDER);
```
Comment (yang-guo-2016) [inline platform-api/src/main/java/com/flow/platform/api/dao/job/JobDaoImpl.java]: QUERY_BUILDER.copy().xxx

**Pull request merged**